### ISSUE TITLE
Sleep as android can no longer connect with diyHue

### ISSUE THREAD

**Issue by Nostras** [diyhue/diyHue#19]
**Describe the bug**
DiyHue used to work fine with Sleep as android.

**To Reproduce**
Simply install sleep as android and attempt to connect it with diyHue, it will hang on attempting to add it.
I tried installing an older version, but with these it still fails.
As diyHue has been updated I assume the problem must lie here.

**Expected behavior**
Well, it should be working.

**Comment by mariusmotea** [comment]
What is this Sleep? An app, a function of Android?

**Comment by Nostras** [comment]
It's an app.
https://play.google.com/store/apps/details?id=com.urbandroid.sleep&hl=en

**Comment by mariusmotea** [comment]
I see. This app use hueManic as hue plugin and this one don't pair anymore with diyhue. From logs a see a successful pair, this app don't like something from the config. I will try to find out what.

**Comment by Nostras** [comment]
Good to hear.
Let me know when it is available and I'll make sure to share my experiences.

**Comment by Nostras** [comment]
I tried it again as someone mentioned it worked here.
https://github.com/diyhue/diyHue/issues/51

The current version is not detected by Sleep ( at all ) and Hue won't connect ( I don't really care about the latter tbh ).

**Comment by Nostras** [comment]
I am running 20181212 and as far as I know that is the latest version. That one doesn't work for me.

**Comment by Nostras** [comment]
Even apkpure says that the latest version is 20181212. How the hell did you get your hands on that one?
https://apkpure.com/sleep-as-android-sleep-cycle-tracker-smart-alarm/com.urbandroid.sleep

Sorry @diyHue for going a bit off topic, but this is bugging me.

**Comment by Nostras** [comment]
Yeah, Sleep as Android only shows a bridge on address 192.168.1.124 as a valid huebridge, that subnet is not even used where I am, everything is 192.168.123.XXX. That also makes it kind of obvious that this is not the fault of DiyHue as far as I can tell.
I can't even bloody enter a static address, dumbing down interfaces ffs.

**Comment by Nostras** [comment]
I dusted off my old phone with a clean sleep as android from the 22nd of January and the one joshuant has. It keeps saying the ip in the previous comment. The DiyHue just doesn't appear at all. I'm once again not sure who to blame, sleep being buggy or sleep as android not properly announcing/responding.

**Comment by Nostras** [comment]
Might wanna update here that it is now passed as an argument ( -> --debug True )
https://github.com/mariusmotea/diyHue/wiki/Debug
Instead of being a variable in the header.

First things first, Sleep as android **cannot** find the server by default, but it **can** when it is running with debug. So I think that's a DiyHue thing.
It also doesn't seem to work very well if I enble debug but pipe stdout to a file. Strange.
Anyways, here's the log.
https://pastebin.com/hyyvzH86
It's stuck at line 55 as you can evidently see by the caret C.

**Comment by cheesemarathon** [comment]
Thanks for spotting the error in the wiki! I have updated the instructions at (ReadTheDocs)[diyhue.rtfd.org] as all our documentation is there now.

Thanks for the debug, the current behaviour is very odd. @mariusmotea can you have a look at this?

**Comment by PhoenixTwoFive** [comment]
Hello, I just want to chime in with a piece of information that I thought might be useful:
I have the aforementioned issues only when trying to link Sleep as Android/ other apps using hueManic to an instance of diyhue running on an Raspberry Pi (Model B). The "countdown bar" on the app's link screen just freezes. When firing up an instance on my PC in Docker it works flawlessly.

I hope this is in some way useful, feel free to tell me if you could use my help in any way with this.

**Comment by Nostras** [comment]
Won't even show up in the bridge list.

If it matters I'm running it on an x64 debian stretch platform.

**Comment by PapACutA** [comment]
If i do a new install of diyHue and don't add any lights or rooms my sleep as android app links succesfull. After i added a light and one room with the original Hue app my sleep as android freezes at requesting / getting the config. So it looks like sleep as android really don't like the configuration. Does someone have an Original Hue Bridge that links succesfully with sleep as android so we can look at its configuration and spot the differences.

**Comment by PapACutA** [comment]
I don't no if it helps but i played a bit with the config.json and after i deleted all scenes my sleep as android linked succesfully. It don't work very well. Without the scenes only one simple dimmable light worked right, my rgb led strips didn't do anything.

**Comment by cheesemarathon** [comment]
@PapACutA Did you get anywhere trying to find the source of the issue here?

**Comment by PapACutA** [comment]
Unfortunately not. I have not really had time to spend more time in the last 2 months either.
But my workaround #86 still works for me.

**Comment by Real-Konai** [comment]
Is there any progress to find this issue?
Have anyone tried this apps with an original Hue Bridge?
I use Sleep as Android and Twilight as well, wich are from the same developer. Maybe it's an issue in that apps itself, if it is, we can do nothing here.
@PapACutA An interesting thing is, your quick and dirty fix works with both apps.

**Comment by juanesf** [comment]
Today I tried to link sleep as Android with diyHue and connected, someone else can try if it works

**Comment by alexyao2015** [comment]
Huematic from the same dev will not pair even with the scene removal patch. We need to look into the scenes returned by diyhue and an actual bridge.

**Comment by juanesf** [comment]
HueManiac and Sleep as Android works for me. after reinstalling the emulator from scratch

**Comment by alexyao2015** [comment]
Really? That must mean that either a recent patch fixed it or changing a setting causes the issue.

**Comment by alexyao2015** [comment]
I just tested it again. With a fresh install without any of the default scenes (ie. no scenes at all in the config), both apps pair. The second you add a scene in, it refuses to pair. Clearly, it doesn't like something about the way the scenes are returned. Something that I believe @PapACutA discovered earlier.

We need to figure out what it doesn't like and fix it. Though its hard for me since I don't have an actual bridge.

**Comment by juanesf** [comment]
I don't know if it affects anything but when I tested the installation and application link it was before the last pull https://github.com/diyhue/diyHue/commit/3470c83b98c10897562652bc2290c13642c9f202

**Comment by alexyao2015** [comment]
Did you have scenes before you paired the app?

**Comment by alexyao2015** [comment]
Alright I think I found a difference between whats returned with diyhue and the actual api in scenes.

**Comment by alexyao2015** [comment]
I believe that I have solved the issue and we can finally put this issue to rest 🎉

**Comment by Nostras** [comment]
I'm kicking this again as the problem has reappeared after upgrading to Android 10 and having to reconnect everything.
I also can't connect through the hue apps, at times it will flash as present very shortly but disappear before I have the chance to connect.

**Comment by alexyao2015** [comment]
Go use the beta version. Believe some new commit caused it. If that doesn't work wipe the config and cert and try again

**Comment by Nostras** [comment]
The latter I already tried, I'll try the beta tomorrow.

**Comment by alexyao2015** [comment]
I've been pretty away from dev for a bit so I haven't validated any of the recent commits to not cause issues but I do have an idea of a commit that's causing issues.

**Comment by Nostras** [comment]
I did the manual install (git clone https://github.com/diyhue/diyHue.git), same issue though.

**Comment by alexyao2015** [comment]
Wrong branch. Make sure you use the beta branch. That's the master branch.

https://github.com/diyhue/diyHue/tree/beta?files=1

**Comment by Nostras** [comment]
Hmm, I did a clone and checkout to beta, but I can't seem to run it.

![image](https://user-images.githubusercontent.com/4203238/71843960-60444980-30c5-11ea-89c6-206281956bf0.png)

**Comment by alexyao2015** [comment]
Looks like you didn't clear the config.

If that still doesn't work try docker. Dependency versions have been updated in newer versions

**Comment by Nostras** [comment]
Hmm, still doesn't seem to work.
This is what I see when I try to find devices with the hue app.
I don't see anything happen when I refresh in sleep as android.
![image](https://user-images.githubusercontent.com/4203238/71927401-aec11900-3195-11ea-85d5-8c3fe081e7f2.png)

**Comment by alexyao2015** [comment]
I'll let it at this, if it's still not working, you did not install it properly. Try things like clearing the config, wiping the os and starting over, or using docker instead. That particular build is known to work properly.

**Comment by Nostras** [comment]
I already tried the docker one, but I'll kick it again.

Aside from that, I tried following the manual instructions from here.
https://diyhue.readthedocs.io/en/latest/getting_started.html
Apparently it isn't conclusive then.

Thanks for helping. If it still won't work I'll create a new issue this weekend or so.

**Comment by alexyao2015** [comment]
No need for a new issue. I'm almost positive its not working because of this commit https://github.com/diyhue/diyHue/commit/444fb781712cc770b68c6cd7401b9dc0f27663ad

You are correct, the manual install instructions are incomplete. I would highly suggest using this build here on docker hub 

docker pull diyhue/core:beta-latest

or this one

docker pull yaoa/diyhue:DEBUG-latest

These two builds are known to be stable (moreso the beta-latest over the debug build) and should not have issues. 

Since HueManic and sleep both use the same pairing protocols and since huemanic was last updated prior to the last commit in those branches, its safe to say that those builds definitely work because I have personally tested it.

**Comment by Nostras** [comment]
I tried to use either, but both inhibit similar behavior.
For the record, I tried this.
![image](https://user-images.githubusercontent.com/4203238/71929151-4ecc7180-3199-11ea-9590-cc41ad38c8b6.png)
Verified it was running by going to the IP of the NAS, finally tried to connect through either the Hue app and Sleep as Android.
Both no luck.

If what I'm doing is correct and you're confident its working on your end, something is going wrong in the network here.

**Comment by alexyao2015** [comment]
So diyhue must have access to port 80 and 443. If your Nas is using either of those ports, it won't work. If the hue app will not connect at all, there is something wrong with your configuration.

**Comment by Nostras** [comment]
Alright Alex,

I dusted off an old raspberry Pi and tried the three docker images.
The debug build and beta-latest both crashloop and can't be ran, sadly.

The official build actually does work, in the Hue app that is (it's progress!)...
Sleep as Android however, still won't bloody run. Egh.

**Comment by alexyao2015** [comment]
First, have you tried docker? Second, did you clear the config and cert like I asked?

Those two builds have been checked and maintained by me so I know every build there works. If you are still having trouble, ill try to start from scratch using those builds and see if I can reproduce these issues. However, it sounds like a config or dependency issue.

**Comment by Nostras** [comment]
Hi alex, could you please tell me _where_ these files are?
In docker I deleted them from /mnt /export or something as I couldn't find any anywhere else.

**Comment by alexyao2015** [comment]
@Nostras Can you create a new post in the discourse regarding this issue? This thread is getting too long and hard to follow. I will be rewriting and validating the new commits soon.

**Comment by alexyao2015** [comment]
@Nostras 
I finally got around to testing this around. 2 findings. 1. It seems the latest build works fine with sleep. 2. The error you were getting was related to the incorrect default config being downloaded. I'll be pushing a fix for that shortly.

**Comment by Nostras** [comment]
Great to hear you tested it and saw the same. I'll regularly check in to see if you've committed anything.

**Comment by alexyao2015** [comment]
I've already committed the fix to beta and a new stable branch at least for docker. Try stable first.

**Comment by Nostras** [comment]
Didn't really work on the Nas (same issue as before) but works fine on the Pi after refreshing 10+ times.

I'd say it's solved, are you seeing the same?

**Comment by alexyao2015** [comment]
I'm not sure if it should have ever worked on the nas due to the port 80 and 443 requirements. Not sure about the 10+ refreshes you mentioned.

**Comment by Nostras** [comment]
Port 80 and 443 are open and free on the NAS. Nonetheless I do not think I can fault DiyHue this issue as it works(?) on the Pi.
On the Pi I get the same behavior as I had on the Nas, it does work, but it does take a lot of refreshes (and luck?).

If you're not seeing this perhaps something on the network is in some way hampering the connection and others are unlikely to see this issue. Ergo not DiyHues fault.

**Comment by alexyao2015** [comment]
I run diyhue in a vm using docker and do not see these issues. As this thread is getting long, please make a new thread on the forums.

**Comment by cheesemarathon** [comment]
This issue has been mentioned on **diyHue**. There might be relevant details there:

https://diyhue.discourse.group/t/diyhue-on-nas-cant-be-detected-works-fine-on-pi/58/1
